### PR TITLE
Add Slither reentrancy check in CI

### PR DESCRIPTION
**Pull request by JulissaDantes**
Adds a new check for the github workflows.

#### PR Checklist (None apply)

- [ ] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -54,3 +54,31 @@ jobs:
...
+          restore-keys: npm-v2-
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip3 install slither-analyzer
```
Comment (frangio) [inline .github/workflows/test.yml:82]: Curious to know more about this, was it necessary to upgrade pip?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -54,3 +54,31 @@ jobs:
...
+      - name: Summary of static ananlysis
```
Comment (frangio) [inline .github/workflows/test.yml:83]: There's a typo in this line.

File: .github/workflows/test.yml (hunk 1 above)
Comment (JulissaDantes) [reply .github/workflows/test.yml:82]: No, I leave it from a local run where I needed to upgrade pip, but after checking its not needed so I removed it

**Comment by Amxx** [comment]
This PR contains a lot of changes that I understand are unrelated (certora in particular).
Are these related to enabling slither?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -54,3 +54,29 @@ jobs:
...
+
+  slither:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+      - run: yarn install
+      - run: yarn compile
```
Comment (frangio) [inline .github/workflows/test.yml:66]: This project doesn't use Yarn. Installing dependencies should be done with npm like the other jobs in this file.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -54,3 +54,29 @@ jobs:
...
+      - uses: actions/cache@v2
+        id: cache
+        with:
+          path: '**/node_modules'
+          key: npm-v2-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: npm-v2-
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
```
Comment (frangio) [inline .github/workflows/test.yml:77]: We're not using the [matrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) and we don't have a `python-version` parameter so this is not necessary.
```suggestion
      - name: Set up Python
        uses: actions/setup-python@v2
```

File: package.json [hunk 5]
```diff
@@ -29,7 +29,8 @@
     "version": "scripts/release/version.sh",
     "test": "hardhat test",
     "test:inheritance": "node scripts/inheritanceOrdering artifacts/build-info/*",
-    "gas-report": "env ENABLE_GAS_REPORT=true npm run test"
+    "gas-report": "env ENABLE_GAS_REPORT=true npm run test",
+    "slither": "npm run clean && slither . --detect reentrancy-eth,reentrancy-no-eth,reentrancy-unlimited-gas"
```
Comment (frangio) [inline package.json:33]: Do you know why `clean` is necessary here?

Comment (JulissaDantes) [reply package.json:33]: If somebody makes a change to fix the reentrancy or just add the diable comment and doesnt compile before pushing the code, slither will check the last compiled contract and show the warning again, unless you clean it before so slither compiles it itself

**Review by frangio — APPROVED** [review]

**Review by BigBlueHundids — COMMENTED** [review]
>.

**Pull request merged**